addons/point_of_sale/models/pos_payment_method.py

```diff
@@ -96,4 +96,7 @@ def write(self, vals):
 
     def copy(self, default=None):
         default = dict(default or {}, config_ids=[(5, 0, 0)])
+        if self.journal_id and self.journal_id.type == 'cash':
+            if 'journal_id' in default and default['journal_id'] == self.journal_id.id:
```

Comment:
I think you can remove this line, the fix won't work with it as the default dict does not contain journal_id most of the time when duplicating. Just say that, if the payment method is cash, the duplicate will not be linked to a journal and the user has to update it himself afterwards

Reply:
@adgu-odoo 

https://github.com/odoo/odoo/blob/beeba3b55e5ac645577288c53d56c63e30643cdc/addons/point_of_sale/tests/common.py#L253-L266 

Should I fix this test case by using create instead of copy, or should I add an additional condition?
```py
if 'journal_id' not in default:
    default.update({'journal_id': False})
``` 
what do you think so?

Reply:
because otherwise test case is failing sorry, I forgot to add this

Reply:
What you did is good for me, I'll r+ it